### PR TITLE
Added convenience methods for matrices in matrices.py

### LINKED ISSUE #16619: convenience methods for matrices
Here are some functions that might be useful to have for working with matrices:
```python
def randMatrixBigDiag(rows, cols=None, lo=0, hi=100, symmetric=False):
    """Return a matrix filled with random values such that no
    value is greater than the values on the diagonals in the same
    row or column containing the value.

    Examples
    ========

    >>> randMatrixBigDiag(4)
    Matrix([
    [20,  6,  7, 12],
    [ 0, 36, 19, 23],
    [17, 15, 43, 20],
    [10,  5,  3, 45]])
    >>> randMatrixBigDiag(4, symmetric=True)
    Matrix([
    [56, 54, 17, 30],
    [54, 87,  5, 17],
    [17,  5, 35, 16],
    [30, 17, 16, 34]])
    >>> randMatrixBigDiag(4, 5)
    Matrix([
    [55, 28, 31, 23],
    [29, 32,  3,  6],
    [12, 16, 75, 42],
    [30,  1, 14, 56]])
    >>> randMatrixBigDiag(4, hi=10, symmetric=True)
    Matrix([
    [6, 5, 1, 2],
    [5, 7, 2, 2],
    [1, 2, 9, 1],
    [2, 2, 1, 5]])
    """
    from random import randint
    cols = rows if cols is None else cols
    d = [randint(lo, hi) for i in range(min(rows, cols))]
    A = diag(*d)
    if symmetric:
        if not len(set(A.shape)) == 1:
            raise ValueError('only a square matrix can be symmetric')
        for i in range(A.rows):
            for j in range(i + 1, A.cols):
                A[j,i] = A[i,j] = randint(0, min(d[i], d[j]) - 1)
    else:
        for i in range(A.rows):
            for j in range(A.cols):
                if i != j:
                    A[i,j] = randint(0, min(d[i], d[j]) - 1)
    return A

def upper(a, diag=0):
    """
    Examples
    ========

    >>> upper(ones(4,2))
    Matrix([
    [1, 1],
    [0, 1],
    [0, 0],
    [0, 0]])
    >>> upper(ones(2,4))
    Matrix([
    [1, 1, 1, 1],
    [0, 1, 1, 1]])
    >>> upper(ones(3), diag=1)
    Matrix([
    [0, 1, 1],
    [0, 0, 1],
    [0, 0, 0]])
    """
    b = Matrix(a)
    d = min(a.rows, a.cols)
    for i in range(d):
        for j in range(i + diag):
            b[i, j] = 0
    for i in range(d, a.rows):
        for j in range(a.cols):
            b[i, j] = 0
    return b

def lower(a, diag=0):
    """
    Examples
    ========

    >>> lower(ones(4), -1)
    Matrix([
    [0, 0, 0, 0],
    [1, 0, 0, 0],
    [1, 1, 0, 0],
    [1, 1, 1, 0]])
    >>> lower(ones(2, 4))
    Matrix([
    [1, 0, 0, 0],
    [1, 1, 0, 0]])
    >>> lower(ones(4, 2))
    Matrix([
    [1, 0],
    [1, 1],
    [1, 1],
    [1, 1]])
    """
    b = Matrix(a)
    i0 = diag + 1
    for i in range(a.rows):
        for j in range(i + i0, a.cols):
            b[i, j] = 0
    return b
```

### PR DESCRIPTION
**Pull request by mohitacecode**
#### References to other Issues or PRs

Fixes #16619

#### Brief description of what is fixed or changed
Added Three convenience methods for matrices
1.) `lower`
2.) `upper`
3.) `randMatrixBigDiag`

#### Release Notes

*   matrices
     *  Added `lower`, `upper`, `randMatrixBigDiag` convenience methods for matrices in matrices.py

### CODE REVIEW

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -3986,3 +3986,117 @@ def classof(A, B):
 def a2idx(j, n=None):
     from sympy.matrices.common import a2idx as a2idx_
     return a2idx_(j, n)
+
+
+def randMatrixBigDiag(rows, cols=None, lo=0, hi=100, symmetric=False):
+    from sympy.matrices.dense import diag
+    """Return a matrix filled with random values such that no
+    value is greater than the values on the diagonals in the same
```
Comment (oscarbenjamin) [inline sympy/matrices/matrices.py]: Imports should not be before docstrings. The docstring will not be saved in `__doc__` if you do that.

I'm not sure that this function should be added as it seems obscure.

Comment (mohitacecode) [reply sympy/matrices/matrices.py]: I shifted imports after docstring.
Should I remove `randMatrixBigDiag` methods? @oscarbenjamin

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -3986,3 +3986,117 @@ def classof(A, B):
...
+                if i != j:
+                    A[i,j] = randint(0, min(d[i], d[j]) - 1)
+    return A
+
+def upper(a, diag=0):
+    """
+    Examples
+    ========
+
+    >>> upper(ones(4,2))
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: There should probably be `from sympy.matrices.matrices import upper` and `from sympy.matrices import ones` before this line. You can test the examples by invoking python and then copying these statements before doing anything else.

**Comment by mohitacecode** [comment]
I have to close this pr due to some unresolvable commits.I will open a new pr with appropriate changes

**Pull request closed without merging**